### PR TITLE
Added matrix type questions to the survey

### LINKED ISSUE #6: Add matrix type questions to the survey

### PR DESCRIPTION
**Pull request by bljubisic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,9 +44,46 @@
...
+      "types": "./dist/md.d.ts",
+      "svelte": "./dist/md.js"
+    },
     ".": {
       "types": "./dist/index.d.ts",
       "svelte": "./dist/index.js"
     }
-  }
+  },
+  "description": "Survey-md is an attempt to simplify creating and using surveys through the code. As a starting point, I used [surv.app project](https://surv.app) which provided the same functionality but on server side. I am trying to offer something like this out of necessity to use similar library for dynamically survey generation using markdown coming from BE. md stands for Markdown as I am using markdown language to describe the surveys. All results are stored in context object and you can easily access them using context[\"question_name\"]. There are couple of rules how to do that:",
```
Comment (netzartist) [inline package.json]: And what are the rules, where does the docu continue?

File: src/lib/SurveyQuestion.svelte
Comment (netzartist) [inline src/lib/SurveyQuestion.svelte:1]: How about switching fully to Typescript?

File: src/lib/SurveyQuestion.svelte [hunk 2]
```diff
@@ -73,6 +77,17 @@
             checked[i] = true;
           }
         }
+      } else if(matrix) {
+        // console.log(Object.keys(checked).filter((key) => (key.includes(i+".") && checked[key] === true)));
```
Comment (netzartist) [inline src/lib/SurveyQuestion.svelte]: console ;)

File: src/lib/SurveyQuestion.svelte [hunk 3]
```diff
@@ -163,26 +196,70 @@
...
+              <SurveyNode
+                node={{ ...child.children[0], type: 'text' }}
+                {context}
+                {next} />
+            </div>
+            {#each matrix as nonused, j}
+              <div class:single={true && !className}>
+                <!-- svelte-ignore a11y-click-events-have-key-events -->
+                <!-- svelte-ignore a11y-no-noninteractive-element-to-interactive-role -->
+                <li
```
Comment (netzartist) [inline src/lib/SurveyQuestion.svelte:223]: What speaks against `input type=radio` (within the `li`s) also to make the a11y checker happier?

**Comment by netzartist** [comment]
I was confused about the behaviour in the UI:
- It is possible to uncheck a "radio" item, normally you cannot do that
- In a row: When I select an item and click on the other one (to select this instead), this is not possible - I first need to uncheck the old one and then I can check the other one

File: src/lib/SurveyQuestion.svelte (hunk 3 above)
Comment (bljubisic) [reply src/lib/SurveyQuestion.svelte:223]: Opted not to make this change at this moment. Should be part of a bigger change of all other types as well

File: src/lib/SurveyQuestion.svelte (hunk 2 above)
Comment (bljubisic) [reply src/lib/SurveyQuestion.svelte]: Done

File: src/lib/SurveyQuestion.svelte
Comment (bljubisic) [reply src/lib/SurveyQuestion.svelte:1]: Done

File: package.json (hunk 1 above)
Comment (bljubisic) [reply package.json]: Done

**Comment by bljubisic** [comment]
> I was confused about the behaviour in the UI:
> 
> * It is possible to uncheck a "radio" item, normally you cannot do that
> * In a row: When I select an item and click on the other one (to select this instead), this is not possible - I first need to uncheck the old one and then I can check the other one

Done, both items

**Comment by netzartist** [comment]
Approved :)

**Pull request merged**